### ISSUE TITLE
Changing learning rate of the optimizer in eager mode

### ISSUE THREAD

**Issue by AakashKumarNain** [tensorflow/tensorflow#23314]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information** 
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 1.11
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory: K80 

**Describe the current behavior**
When using `tf.keras` with `eager` execution enabled, I don't have the control over learning rate of the optimizer i.e. I am unable to change the learning rate on the fly. If you are using pure `keras` with no `eager` mode, then you can change the learning rate of the keras optimizer in the following way 

`K.set_value(self.mode.optimizer.lr, value)`

But as the `keras` optimizers aren't supported in `eager` mode, I don't have any control over the learning rate of my optimizer.

**Describe the expected behavior**
The expected behavior is that I should be able to change the learning rate of my optimizer after a batch ends or after an epoch ends

**Comment by random-forests** [comment]
Here's a quick example of setting the LR using a callable, w/ the Gradient Tape way I mentioned on Twitter: https://colab.research.google.com/drive/10OIcUdwRiHbQK4p32MEABG_G4_D44YKH

This probably isn't quite what you meant -- could you add a code snippet showing the behavior you'd like?

**Comment by AakashKumarNain** [comment]
Yes, that can be another way to tweak the LR within batches/epochs. If I don't use `tf.keras` and `eager`, and switch to pure `Keras` instead with `tf` backend., I can do this in a much cleaner way. Take a look at this: https://colab.research.google.com/drive/1I-vvsD4zhBUxcMh6LEadqUOF9sVgHy2T#scrollTo=t9ZHnrWefHuq

**Comment by AakashKumarNain** [comment]
Yes and No. But Josh and Francois have made it clear that the optimizers are going to be merged in TF 2.0 very soon.

**Comment by harshini-gadige** [comment]
@AakashKumarNain  - Do you want to keep this open until TF 2.0 ? My suggestion is to open a new issue(if it still exists after TF 2.0) with the latest data.
